### PR TITLE
Fixes failing tests; reorganized test files (PHPUnit 10.x related)

### PR DESCRIPTION
**Pull request by k00ni**
Since version 10, PHPUnit complains about:

1) Your XML configuration validates against a deprecated schema. Migrate your XML configuration using "--migrate-configuration"!

2) Class AltAutoloadTest cannot be found in
D:\a\pdfparser\pdfparser\tests\AltAutoloadTest.php

3) Class Tests\Smalot\PdfParser\Performance\Test\AbstractPerformanceTest cannot be found in
D:\a\pdfparser\pdfparser\tests\Performance\Test\AbstractPerformanceTest.php

4) Class
Tests\Smalot\PdfParser\Performance\Test\DocumentDictionaryCacheTest cannot be found in
D:\a\pdfparser\pdfparser\tests\Performance\Test\DocumentDictionaryCacheTest.php

The first one is easy to solve.

But the other 3 are confusing, because PHPUnit 9 did not complain at all. PHPUnit v10 doesn't like abstract Test classes or files (containing Test in the name), which contain no class at all. Source: https://github.com/sebastianbergmann/phpunit/issues/4979

To mitigate these warnings and to make maintenance easier in the future, I reorganized our test files and put them in different folders. PHPUnit tests are separated from our other tests.

#581 and #582 will be processed after this one is merged.

### CODE REVIEW

**Review by j0k3r — COMMENTED** [review]
Maybe we should lock the PHPUnit version, see https://thephp.cc/articles/the-death-star-version-constraint

**Comment by k00ni** [comment]
> Maybe we should lock the PHPUnit version, see https://thephp.cc/articles/the-death-star-version-constraint

I would advise against that, because it only delays required changes. I prefer seeing problems as soon as possible. Maybe there is some compromise here?

**Comment by j0k3r** [comment]
Well, it's a way to do it.
Keeping unlocked version means when PHPUnit 11 will be released someone will have to quickly fix issues (if any) otherwise all PRs will fail.
Keeping a locked version avoid that problem. And then, we can smoothly update PHPUnit later without hurry

**Comment by k00ni** [comment]
Good points, I implemented a lock.

What doyou think about the rest?

File: dev-tools/composer.json [hunk 1]
```diff
@@ -4,6 +4,6 @@
         "friendsofphp/php-cs-fixer": "^3",
         "phpstan/phpstan": "^1",
         "phpstan/phpstan-phpunit": "^1",
-        "phpunit/phpunit": ">=7.5"
+        "phpunit/phpunit": ">=7.5 <10.0"
```
Comment (j0k3r) [inline dev-tools/composer.json:7]: Shouldn't this be like that?
```suggestion
        "phpunit/phpunit": ">=7.5 <11.0"
```

**Review by j0k3r — COMMENTED** [review]
The rest looks ok 👍🏼

**Pull request merged**